### DISCUSSION TITLE
Prisma data model MySQL, TEXT, JSON not supported ?

### DISCUSSION THREAD

**Discussion by amitdhiman000** [prisma/prisma#2741] [Q&A] [answered]
Prisma data model MySQL, 
String maps to  VARCHAR which has limit on number of characters (I think it is 191)
Json maps to LONGTEXT which is not actual JSON.

Can we expect support for TEXT, JSON in prisma 2.

I am working on an application where we need both.
otherwise we will have to drop prisma.

**Comment by ryands17** [comment, marked as answer]
Hey @amitdhiman000 :wave: 
Currently these types are directly not supported by Prisma Migrate, and you can track the status of this request [here](https://github.com/prisma/prisma/issues/446).

As a workaround, you would need to create the types directly in the DB and then run `prisma introspect` to get the models.

**Reply by amitdhiman000** [reply]
@ryands17 thanks.

**Comment by ludralph** [comment]
Hi @amitdhiman000,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
